blocks/embed/embed.js

```diff
@@ -128,11 +128,13 @@ function decorateBlockEmbed($block, url) {
   }
 }
 
-export default function decorate($block) {
-  const $a = $block.querySelector('a[href]');
-  $block.textContent = '';
-  if ($a) {
-    const url = new URL($a.href.replace(/\/$/, ''));
-    decorateBlockEmbed($block, url);
-  }
+export default function decorate(block) {
+  const link = block.querySelector('a').href;
+  block.textContent = '';
+  const observer = new IntersectionObserver((entries) => {
+    if (entries.some((e) => e.isIntersecting)) {
+      decorateBlockEmbed(block, new URL(link));
```

Comment:
@davidnuescheler don't you need to remove the observer, once loaded?

Reply:
Since the embedding needs to happen only one, perhaps we can `unobserve` once its done.

Reply:
ah, i would do a disconnect... sorry forgot that...